src/object_async/hello_async.cpp

```diff
@@ -0,0 +1,227 @@
+#include "hello_async.hpp"
+
+#include <exception>
+#include <stdexcept>
+#include <iostream>
+#include <map>
+
+// If this was not defined within a namespace, it would be in the global scope.
+namespace object_async {
+
+  // Custom constructor, assigns custom name passed in from Javascript world.
+  // This constructor uses member init list via the semicolon, aka "direct initialization" 
+  // which is more efficient than using assignment operators.
+  HelloObjectAsync::HelloObjectAsync(std::string name) : 
+    name_(name) {}
+
+  // Triggered from Javascript world when calling "new HelloObjectAsync()" or "new HelloObjectAsync(name)"
+  NAN_METHOD(HelloObjectAsync::New) {
+    if (info.IsConstructCall())
+    {
+        try
+        {   
+            if (info.Length() >= 1) {
+              if (info[0]->IsString()) 
+              {
+                std::string name = *v8::String::Utf8Value(info[0]->ToString());
+                auto *const self = new HelloObjectAsync(name);
+                self->Wrap(info.This());
+              }
+              else
+              {
+                return Nan::ThrowTypeError(
+                    "arg must be a string");
+              }
+            }
+            else {
+                auto *const self = new HelloObjectAsync();
+                self->Wrap(info.This());
+            } 
```

Comment:
I think a more realistic thing would be to throw an error if no argument is passed. Currently we default to `world` below. But I think instead we should throw a JS error here if no `name` arg is passed. What do you think?